### PR TITLE
tapfreighter: add porter state SendStateResumeTransfer

### PR DESCRIPTION
**Pull request by ffranr**
In this state the porter determines the next appropriate state for resuming the processing of a pending package. The next state is determined based on whether the package anchor tx has been mined.

I haven't tested this change yet, but early feedback is welcomed.

### CODE REVIEW

**Comment by jharveyb** [comment]
I think this is missing a jump into the `ResumeTransfer` state.

Maybe set that here:

https://github.com/lightninglabs/taproot-assets/blob/f4c0e8e70331da9171ff8af38ab1099d5df3bd49/tapfreighter/parcel.go#L188

And then the resume logic you added in the last PR should take care of the rest.

**Review by guggero — COMMENTED** [review]
Left a few comments. I'm not sure I understand the reason for this change. If a transaction was valid/accepted before the restart, it should be no problem to re-broadcast it. If it is still valid (or has even been mined in the meantime) no error should be returned. So I think we can still just resume from the `SendStateBroadcast` state for any package on daemon restart.

File: tapfreighter/chain_porter.go [hunk 1]
```diff
@@ -217,6 +217,52 @@ func (p *ChainPorter) assetsPorter() {
...
+		anchorTxHeightHint = outboundPkg.AnchorTxHeightHint
+	)
+	log.Infof("Resuming package transfer: transfer_txid=%v", txHash)
+
+	// Attempt to identify if the anchor tx has been mined.
+	//
+	// At this point we are not waiting for the anchor tx to be mined as it
+	// might not have been broadcast yet. We will therefore set a minimal
+	// timeout for the tx confirmation.
+	ctx, cancel := p.WithCtxQuitCustomTimeout(3 * time.Second)
```
[the remark this answers is not in the document]
Comment (guggero) [reply tapfreighter/chain_porter.go:235]: I just tried this out in regtest, and if I re-broadcast a transaction that has already confirmed (the identical TX), I get `Transaction already in block chain`, which I think is treated as a non-error by the chain handler. If this code is used: https://github.com/btcsuite/btcwallet/blob/6c69ff82eee4298154a5dc404a87c1f0f58495eb/wallet/wallet.go#L3690

Not sure what happens if the node is pruned or has no TX index though.

[the remark this answers is not in the document]
Comment (jharveyb) [reply tapfreighter/chain_porter.go:235]: Explained scenario more verbosely here:

https://github.com/lightninglabs/taproot-assets/issues/332#issuecomment-1593325969

**Comment by ffranr** [comment]
I'm not sure if I should close this PR or not anymore. But regardless, I think we should merge https://github.com/lightninglabs/taproot-assets/pull/365 first and then I will rebase this PR on top.

I figured that I would keep these PRs separate because I'm pretty confident we will want to merge https://github.com/lightninglabs/taproot-assets/pull/365 once its gone through the normal review cycle.

File: tapfreighter/chain_porter.go [hunk 2]
```diff
@@ -938,8 +990,9 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 			return nil, err
 		}
 
-		currentPkg.TransferTxConfEvent = confEvent
-		currentPkg.SendState = SendStateStoreProofs
+		// Update the package porter state given that the anchor tx was
+		// confirmed.
+		p.setStateAnchorTxConf(&currentPkg, confEvent)
```
[the remark this answers is not in the document]
Comment (ffranr) [reply tapfreighter/chain_porter.go]: Sure, no problem.

**Comment by guggero** [comment]
I think if we have an itest that tests both scenario 2 and 3 (so 2: "transfer, stop daemon, no mining, resume" and 3: "transfer, stop daemon, mine a block, resume") and both of them succeed, we don't need this PR anymore IMO.
So I agree that we should get in #365 (which addresses 2 as far as I can tell).

**Comment by ffranr** [comment]
Closed in favour of https://github.com/lightninglabs/taproot-assets/pull/365

**Pull request closed without merging**